server/src/main/java/keywhiz/ServiceModule.java

```diff
@@ -62,18 +66,18 @@ public ServiceModule(KeywhizConfig config, Environment environment) {
     // Initialize the BouncyCastle security provider for cryptography support.
     BouncyCastle.require();
 
+    bind(ClientCertificateFilter.class).toProvider(ClientCertificateFilter::new);
     bind(Clock.class).toInstance(Clock.systemUTC());
-
-    install(new CookieModule(config.getCookieKey()));
-    install(new CryptoModule(config.getDerivationProviderClass(), config.getContentKeyStore()));
-
     bind(CookieConfig.class).annotatedWith(SessionCookie.class)
         .toInstance(config.getSessionCookieConfig());
+    bind(SecurityHeadersFilter.class).toProvider(SecurityHeadersFilter::new);
+    bind(SessionMeResource.class).toProvider(SessionMeResource::new);
 
-    // TODO(justin): Consider https://github.com/HubSpot/dropwizard-guice.
-    bind(Environment.class).toInstance(environment);
-    bind(Configuration.class).toInstance(config);
-    bind(KeywhizConfig.class).toInstance(config);
+    install(new ContextModule(config, environment));
+    install(new CookieModule(config.getCookieKey()));
+    install(new CryptoModule(config.getDerivationProviderClass(), config.getContentKeyStore()));
+    install(new DaoModule());
+    install(new StrictGuiceModule());
   }
```

Comment:
This is mostly just organization:

1. All bind() statements are in a single block, likewise install()s
2. Bindings for Environment, Configuration, and KeywhizConfig are moved to a dedicated module (ContextModule)
3. Dao-related bindings are in their own module
4. In order to circumvent some subtle bugs that came up in testing (Guice will, by default, instantiation class with no-arg constructors, which masked an issue where KeywhizConfig wasn't actually loading properly), we've turned on some stricter binding settings which are located inside StrictGuiceModule. As a consequence, classes with no-argument constructors that we want injected (ClientCertificateFilter, SessionMeResource, etc.) need explicit bindings.